index.js

```diff
@@ -515,6 +508,16 @@ studio.internal = (function(proto) {
       app.makeSetterRequest(id, lastInfo.value_type, value, timestamp);
       //when offline must queue or update pending set request and call set callbacks ...???
     };
+
+    this.async._makeGetterRequest = function() {
+      if (valueSubscriptions.length > 0) {
+        var maxFs = Math.max.apply(Math, valueSubscriptions.map(v => v[1]));
+        var maxSampleRate = Math.max.apply(Math, valueSubscriptions.map(v => v[2]));
```

Comment:
Technically sample_rate of 0 is the highest sample rate, so Math.max is not sufficient.

Reply:
was approved in python client

Reply:
Should be fixed there as well then